### PR TITLE
[ci/bisect] allow sanity check to run multiple times per commit

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
Currently even when users specify to run multiple times per commit, sanity_check only run once per commit. To be consistent and to accommodate for flaky tests, allow sanity check to run multiple times per commit.

#auto_complete_by_autopilot

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests

### CODE REVIEW

File: release/ray_release/scripts/ray_bisect.py [hunk 1]
```diff
@@ -102,15 +106,19 @@ def _sanity_check(test: Test, passing_revision: str, failing_revision: str) -> b
         f"Sanity check passing revision: {passing_revision}"
         f" and failing revision: {failing_revision}"
     )
-    outcomes = _run_test(test, [passing_revision, failing_revision])
-    return (
-        outcomes[passing_revision][0] == "passed"
-        and outcomes[failing_revision][0] != "passed"
+    outcomes = _run_test(test, [passing_revision, failing_revision], run_per_commit)
+    passed_on_passing_revision = all(
+        outcome == "passed" for outcome in outcomes[passing_revision].values()
```
Comment (aslonnie) [inline release/ray_release/scripts/ray_bisect.py:111]: (nit) maybe:

````
if not all(map(lambda x: x == "passed", outcomes[passing_revision].values())):
    return False
return all(map(lambda x: x != "passed", outcomes[failing_revision].values())):
````

which is:
- slightly shorter (well, you can also achieve the same thing by renameing `outcome` to `x` I guess)
- shortcut if the first condition is False, without calculating the second one

**Review by aslonnie — APPROVED** [review]
lgtm; just some nits..

File: release/ray_release/scripts/ray_bisect.py (hunk 1 above)
Comment (can-anyscale) [reply release/ray_release/scripts/ray_bisect.py:111]: fun fact, the existing code is generated by github copilot ;))

**Review by krfricke — APPROVED** [review]
LGTM

**Review by matthewdeng — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/ufc/">UFC</a> on <a href="https://giphy.com/gifs/ufc-sports-sport-ufc-285-mma-8FYAJlfHHFeZFYMYmU">GIPHY</a></div>

**Pull request merged**